docs/programming/tilebuffer.md

```diff
@@ -0,0 +1,285 @@
+  keyCreate(); // We'll use keyboard
+	joyOpen();
+
+  s_pView = viewCreate(0, TAG_END);
+
+  // Viewport for score bar - on top of screen
+	s_pVpScore = vPortCreate(0,
+		TAG_VPORT_VIEW, s_pView,
+		TAG_VPORT_BPP, GAME_BPP,
+		TAG_VPORT_HEIGHT, SCORE_HEIGHT,
+	TAG_END);
+
+	s_pScoreBuffer = simpleBufferCreate(0,
+		TAG_SIMPLEBUFFER_VPORT, s_pVpScore,
+		TAG_SIMPLEBUFFER_BITMAP_FLAGS, BMF_CLEAR,
+		TAG_SIMPLEBUFFER_BOUND_WIDTH, 320,
+		TAG_SIMPLEBUFFER_BOUND_HEIGHT, SCORE_HEIGHT,
+	TAG_END);
+
+	// Now let's do the same for main playfield where the scrolling happen
+	s_pVpMain = vPortCreate(0,
+		TAG_VPORT_VIEW, s_pView,
+		TAG_VPORT_BPP, GAME_BPP,
+	TAG_END);
+
+  // Load tiles
+	s_pTiles = bitmapCreateFromPath("data/overworld.bm", 0);
+
+	logWrite("Tiles Loaded!\n");
+
+  // Init your scrolling tile buffer 
+	s_pMainBuffer = tileBufferCreate(0,
+		TAG_TILEBUFFER_VPORT, s_pVpMain,  // Select View Port
+		TAG_TILEBUFFER_BITMAP_FLAGS, BMF_CLEAR | BMF_INTERLEAVED,  // If you put interleaved there, it should be also in your cmakelist part.
+		TAG_TILEBUFFER_BOUND_TILE_X, TILE_MAP_SIZE_X, // Map size in number of tile
+		TAG_TILEBUFFER_BOUND_TILE_Y, TILE_MAP_SIZE_Y, // Map size in number of tile
+		TAG_TILEBUFFER_IS_DBLBUF, 1, // Activatey double buffering
+		TAG_TILEBUFFER_TILE_SHIFT, 4, //Size of tile, given in bitshift. Set to 4 for 16px, 5 for 32px, etc. Mandatory.
+		TAG_TILEBUFFER_REDRAW_QUEUE_LENGTH, 200, // cursor between handling fast scrolling versus your amiga power
+		TAG_TILEBUFFER_CALLBACK_TILE_DRAW, onTileDraw, // To be documented
+		TAG_TILEBUFFER_TILESET, s_pTiles,
+	TAG_END);
+
+
+  // Load and apply palette
+  paletteLoadFromPath("data/overworld.plt", s_pPalette, 1 << GAME_BPP);
+	memcpy(s_pVpScore->pPalette, s_pPalette, sizeof(s_pVpScore->pPalette));
+	memcpy(s_pVpMain->pPalette, s_pPalette, sizeof(s_pVpMain->pPalette));
+
+
+	loadMap();
+
+  // End of IO let's display somme stuff - unlaod OS
+	systemUnuse();
+
+	logWrite("Camera Reset\n");
+	cameraReset(s_pMainBuffer->pCamera, 0, 0, s_uwMapTileWidth*16, s_uwMapTileHeight * 16, 1);
+
+  // Initial draw
+	tileBufferRedrawAll(s_pMainBuffer);
+
+  // Set initiale camera
+	cameraSetCoord(s_pMainBuffer->pCamera, g_uCameraX, g_uCameraY);
+
+	// Load & Display the view
+	viewLoad(s_pView);
+```
+
+And in your `gameGsDestroy` method to unload memory :
+
+```c
+void gameGsDestroy(void) {
+  systemUse();
+  bitmapDestroy(s_pTiles);
+  viewDestroy(s_pView);
+  keyDestroy();
+  joyClose(); 
+}
+```
+
+Finally let's control our scrolling in `gameGsLoop`:
+
+```c
+  // Handle keyboard and joystick events
+  keyProcess();
+  joyProcess();
+  //[...]//
+  // Let's control your scrolling
+  // Q for speeding up (A in azerty keyboard)
+  // W for speeding up (Z in azerty keyboard)
+  WORD wDx = 0, wDy = 0;
+	if(keyCheck(KEY_Q)) {
+		g_uCameraSpeed+=1;
+	}
+	if(keyCheck(KEY_W)) {
+		if(g_uCameraSpeed>1) {
+			g_uCameraSpeed-=1;
+		}
+	}
+  // Joy direction to move
+	if(joyCheck(JOY1_UP)) {
```

Comment:
perhaps it's worth to add `|| keyCheck(KEY_UP)` etc?

Reply:
Yep good ID.